registry/api/protocol/eureka/instance_handler.go

```diff
@@ -341,13 +356,86 @@ func (routes *Routes) getInstance(w rest.ResponseWriter, r *rest.Request) {
 		routes.logger.WithFields(log.Fields{
 			"namespace": r.Env[env.Namespace],
 			"error":     err,
-		}).Warn("Failed to encode instance")
+		}).Warnf("Failed to encode instance %s", uid)
 
 		i18n.Error(r, w, http.StatusInternalServerError, i18n.ErrorEncoding)
 		return
 	}
 }
 
+func (routes *Routes) getInstanceByID(w rest.ResponseWriter, r *rest.Request) {
+	iid := r.PathParam(RouteParamInstanceID)
+	if iid == "" {
+		routes.logger.WithFields(log.Fields{
+			"namespace": r.Env[env.Namespace],
+			"error":     "instance id is required",
+		}).Warn("Failed to query instance")
+
+		i18n.Error(r, w, http.StatusBadRequest, i18n.ErrorInstanceIdentifierMissing)
+		return
+	}
+	uid := buildUniqueInstanceID("", iid)
+
+	catalog := routes.catalog(w, r)
+	if catalog == nil {
+		routes.logger.WithFields(log.Fields{
+			"namespace": r.Env[env.Namespace],
+			"error":     "catalog is nil",
+		}).Errorf("Failed to query instance %s", uid)
+
```

Comment:
Need a corresponding `i18n.Error()` too. There's a "namespace not found"-style error string defined.